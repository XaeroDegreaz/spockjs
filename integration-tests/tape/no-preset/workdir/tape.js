import test from 'tape-catch';

test('successful test', t => {
  expect: 1 === 1;
  t.end();
});

test('failing test', t => {
  let x;
  const double = n => n * 2;

  when: x = 3;
  then: x === double(2);
  t.end();
});
