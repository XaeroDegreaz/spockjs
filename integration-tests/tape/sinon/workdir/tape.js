import * as sinon from 'sinon';
import test from 'tape-catch';

test('stubbing', t => {
  const s = sinon.stub();
  stub: s() >> 42;
  t.ok(s() === 42);
  t.end();
});

test('mocking', t => {
  const obj = { method: () => {} };

  const mock = sinon.mock(obj);
  mock: 2 * mock.method();

  obj.method();
  obj.method();

  verify: mock;

  t.end();
});

test('failing verification', t => {
  const obj = { method: () => {} };

  const mock = sinon.mock(obj);
  mock: 2 * mock.method();

  obj.method();

  verify: mock;

  t.end();
});
