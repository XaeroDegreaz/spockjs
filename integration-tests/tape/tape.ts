import { sync as run } from 'cross-spawn';
import { resolve } from 'path';
import { env } from 'process';
import TapParser from 'tap-parser';

import {
  babelJitEnv,
  modulePath,
  requireBabelJitArgs,
  requireTypescriptJitArgs,
  typescriptJitEnv,
} from '../utils';

const tapeCli = resolve(modulePath('tape'), 'bin', 'tape');

export const runTape = (
  cwd: string,
  cb: (status: number, fail: number, failures: any) => void,
) => {
  const { status, stdout } = run(
    'node',
    [tapeCli, ...requireBabelJitArgs, ...requireTypescriptJitArgs, 'tape.js'],
    {
      cwd,
      env: {
        ...env,
        ...babelJitEnv,
        ...typescriptJitEnv,
      },
    },
  );

  // no assertion on pass count possible at the moment,
  // will be 0 because tape only reports assertions, not test cases
  new TapParser(({ fail, failures }) => cb(status, fail, failures)).end(stdout);
};
