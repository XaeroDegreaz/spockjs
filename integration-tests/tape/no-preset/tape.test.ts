import { resolve } from 'path';

import { runTape } from '../tape';

// mark implicit dependencies for jest
() => require('./workdir/tape.js') && require('./workdir/package.json');

const cwd = resolve(__dirname, 'workdir');

test('produces correct output', done => {
  runTape(cwd, (status, fail, [{ name }]) => {
    expect: {
      status === 1;
      fail === 1;
    }

    expect(name).toMatchSnapshot();

    done();
  });
});
