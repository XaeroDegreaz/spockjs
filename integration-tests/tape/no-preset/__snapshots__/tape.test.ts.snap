// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`produces correct output 1`] = `"AssertionError [ERR_ASSERTION]: (x === double(2)) is not truthy # tape.js:13 _assert(x === double(2), \\"(x === double(2)) is not truthy\\") | | | | | 4 3 false [number] double(2) => 4 [number] x => 3"`;
